bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/handler/Utils.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.evcc.internal.handler;
+
+import static org.openhab.binding.evcc.internal.EvccBindingConstants.NUMBER_DIMENSIONLESS;
+import static org.openhab.binding.evcc.internal.EvccBindingConstants.NUMBER_ELECTRIC_CURRENT;
+import static org.openhab.binding.evcc.internal.EvccBindingConstants.NUMBER_EMISSION_INTENSITY;
+import static org.openhab.binding.evcc.internal.EvccBindingConstants.NUMBER_ENERGY;
+import static org.openhab.binding.evcc.internal.EvccBindingConstants.NUMBER_LENGTH;
+import static org.openhab.binding.evcc.internal.EvccBindingConstants.NUMBER_POWER;
+import static org.openhab.binding.evcc.internal.EvccBindingConstants.NUMBER_TIME;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+
+/**
+ * The {@link Utils} provides utility methods
+ *
+ * @author Marcel Goerentz - Initial contribution
+ */
+@NonNullByDefault
+public class Utils {
+
+    private static final Map<String, @Nullable Unit<?>> UNIT_MAP = new HashMap<>();
```

Comment:
Any need for value to be `@Nullable`?

Reply:
This was something with the compiler. I will check it once again.